### ISSUE TITLE
.log_weight variable not hidden

### ISSUE THREAD

**Issue by n-kall** [stan-dev/posterior#88]
It seems that unlike `.chain`, `.iteration` and `.draw`, `.log_weight` is not a hidden variable.
```
x <- example_draws()
weights <- rexp(ndraws(x))
x <- weight_draws(x, weights = weights)
variables(x)
```

```
 [1] "mu"          "tau"         "theta[1]"    "theta[2]"    "theta[3]"   
 [6] "theta[4]"    "theta[5]"    "theta[6]"    "theta[7]"    "theta[8]"   
[11] ".log_weight"
```
This leads to unexpected behavior such as counting it as a variable in `nvariables()` and summarising it with `summarise_draws()`.

**Comment by paul-buerkner** [comment]
Yeah, that is a good point. I am not yet sure what the intended behavior should be to be honest as `.log_weight` has a somewhat different purpose than `.chain` etc.  @jgabry and @mjskay what is your opinion on hiding vs. not hiding `.log_weight`?

**Comment by jgabry** [comment]
Yeah good question. This seems tricky because, like Paul was saying, `.log_weight` is a somewhat unique case. The chain, iteration, etc., are clearly metadata unlike the weights. But the weights also aren't regular variables. So they're somewhere in between. I think I would lean towards hiding them, but I don't have a strong opinion on it. 

What do others think?

**Comment by paul-buerkner** [comment]
I would also tend to hide it. This also has the advantages that it won't be dropped automatically when subsetting draws after variables (without selecting `.log_weight`), which I think is desrible.

If we decide to go that route, we need to extend our meta-columns framework to apply to all draws formats not just to `draws_df` but that should be doable.

**Comment by paul-buerkner** [comment]
Resolved via #91
